### PR TITLE
Improve usability and performance in user api tracing

### PR DESCRIPTION
**Pull request by Anoukh**

### CODE REVIEW

File: stdlib/observability/ballerina-trace/src/main/java/org/ballerinalang/observe/trace/Utils.java [hunk 1]
```diff
@@ -46,4 +51,17 @@ private Utils() {
         }
         return returnMap;
     }
+
+    static BStruct createSpanStruct(Context context, String spanId, String serviceName,
```
Comment (keizer619) [inline stdlib/observability/ballerina-trace/src/main/java/org/ballerinalang/observe/trace/Utils.java]: @Anoukh; shouldn't we make these public?

File: bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java [hunk 2]
```diff
@@ -42,6 +42,10 @@
         return spans.get(spanId);
     }
 
+    Map<String, SpanContext> getSpanContext(String spanId) {
```
Comment (nirmal070125) [inline bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java]: Wrong method name.

File: bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java [hunk 3]
```diff
@@ -50,11 +54,12 @@ void addSpan(String spanId, Map<String, Span> spanMap) {
...
-    void addSpanContext(String spanId, Map<String, SpanContext> spanContextMap) {
-        spanContexts.put(spanId, spanContextMap);
-    }
-
     Map<String, Span> closeSpan(String spanId) {
+        spanContexts.remove(spanId);
         return spans.remove(spanId);
     }
+
+    public void addSpanContext(String spanId, Map<String, SpanContext> spanContextMap) {
```
Comment (nirmal070125) [inline bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java]: Wrong method name.

File: bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java (hunk 2 above)
Comment (Anoukh) [reply bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java]: Fixed in 4ffbfc6ad9a11cc24598f426352bc7c86b4b4724

File: bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java (hunk 3 above)
Comment (Anoukh) [reply bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java]: Fixed in 4ffbfc6ad9a11cc24598f426352bc7c86b4b4724

**Pull request merged**